contracts/strategies/aura/base/AuraStrategyAuraBAL.sol

```diff
@@ -0,0 +1,624 @@
+  address[] public poolAssets;
+  address[] public rewardTokens;
+  address[] public swapAssets;
+  bytes32[] public swapPoolIds;
+
+  constructor() public BaseUpgradeableStrategyUL() {
+    assert(_AURA_DEPOSIT_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.auraDeposit")) - 1));
+    assert(_BALANCER_POOLID_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.balancerPoolId")) - 1));
+    assert(_DEPOSIT_TOKEN_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.depositToken")) - 1));
+    assert(_DEPOSIT_ARRAY_INDEX_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.depositArrayIndex")) - 1));
+    assert(_HODL_RATIO_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.hodlRatio")) - 1));
+    assert(_HODL_VAULT_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.hodlVault")) - 1));
+    assert(_SWAP_NTOKENS_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.swapNTokens")) - 1)); 
+    assert(_DEPOSIT_NTOKENS_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.depositNTokens")) - 1)); 
+  }
+
+  function initializeBaseStrategy(
+    address _storage,
+    address _underlying,
+    address _vault,
+    address _rewardPool,
+    address _auraDeposit,
+    bytes32 _balancerPoolID,
+    address _depositToken,
+    uint256 _depositArrayPosition,
+    uint256 _hodlRatio
+  ) public initializer {
+
+    // calculate profit sharing fee depending on hodlRatio
+    uint256 profitSharingNumerator = 300;
+    if (_hodlRatio >= 3000) {
+      profitSharingNumerator = 0;
+    } 
+    else if (_hodlRatio > 0) {
+      // (profitSharingNumerator - hodlRatio/10) * hodlRatioBase / (hodlRatioBase - hodlRatio)
+      // e.g. with default values: (300 - 1000 / 10) * 10000 / (10000 - 1000)
+      // = (300 - 100) * 10000 / 9000 = 222
+      profitSharingNumerator = profitSharingNumerator.sub(_hodlRatio.div(10)) // subtract hodl ratio from profit sharing numerator
+                                    .mul(hodlRatioBase) // multiply with hodlRatioBase
+                                    .div(hodlRatioBase.sub(_hodlRatio)); // divide by hodlRatioBase minus hodlRatio
+    }
+
+    BaseUpgradeableStrategyUL.initialize(
+      _storage,
+      _underlying,
+      _vault,
+      _rewardPool,
+      weth,
+      profitSharingNumerator,  // profit sharing numerator
+      1000, // profit sharing denominator
+      true, // sell
+      0, // sell floor
+      12 hours, // implementation change delay
+      address(0x7882172921E99d590E097cD600554339fBDBc480) //UL Registry
+    );
+
+    address _lpt = IAuraBaseRewardPool(_rewardPool).stakingToken();
+    require(_lpt == underlying(), "Pool Info does not match underlying");
+    _setSwapNTokens(swapAssets.length);
+    _setDepositNTokens(poolAssets.length);
+    _setAuraDeposit(_auraDeposit);
+    _setDepositArrayIndex(_depositArrayPosition);
+    _setBalancerPoolId(_balancerPoolID);
+    _setDepositToken(_depositToken);
+    setUint256(_HODL_RATIO_SLOT, 1000);
```

Comment:
Seeing this now, not sure if the same problem is in the other bases, but we should set it with the value of `_hodlRatio`, not hardcode the value

Reply:
Fixed. Thanks for the feedback! I saw this is the ConvexUL strategy, not sure if there is a reason that I am missing for the different implementation.

Reply:
Hey, found this too, must have been a mistake on my side when working on the ConvexUL strategy